Move docs into subdirectory to make structor work (#656)
hedgedoc/hedgedoc: docs/content/dev/documentation.md

## Writing

All documentation files are found in the `docs/content` directory of the [hedgedoc/hedgedoc repo](https://github.com/hedgedoc/hedgedoc). These files are just normal markdown files with nothing special about them.

The configuration for mkdocs lies in the `docs` folder in a file called `mkdocs.yml`. With that file the theme and menu - amoung others - can be configured.
**Please note:** Any new files need to be linked to by other files or put in the navigation or the files will be very hard to find on the documentation website.

## Building

To build the documentation locally you need to perform the following steps:

0. make sure you have python3 installed
1. Go into the `docs` folder
2. install all the dependencies possibly with a [venv](https://docs.python.org/3/library/venv.html):  
`pip install -r requirements.txt`

> 3. start the mkdocs dev server `mkdocs serve` or let it build the documentation `mkdocs build`

Reviewer: As we have full sentences in this document and not only bullet-points, I'd suggest capitalizing the first letter of each point and end it with a dot.

> 1. make sure you have python3 installed

Reviewer 2: There should be a 
0. cd docs
?

The necessary Dockerfile and version menu template and also the github action to build the whole documentation can be found in the [docs.hedgedoc.org repo](https://github.com/hedgedoc/docs.hedgedoc.org). This repo is also used to deploy the actuall website to github.io.

> Messor Structor builds and deploys the documentation by finding all branches that follow the pattern `v*` and build each one seperatly via `mkdocs` by first installing the dependencies from `requirements.txt` and then running mkdocs. Afterwards the menu go template is used to include a version switcher in the theme and make all the different versions available.

Reviewer: ```suggestion
Messor Structor builds and deploys the documentation by finding all branches that follow the pattern `v*`. For each branch the docs are generated seperatly by first installing the dependencies from `requirements.txt` and then running mkdocs. Afterwards the menu go template is used to include a version switcher in the theme and make all the different versions available.
```

The long "this is done by a for each done by b" sentence structure is quite hard to read. Maybe this sounds easier to understand?
